receiver/dockerstatsreceiver/metric_helper.go

```diff
@@ -71,3 +76,54 @@ func calculateMemoryPercent(limit uint64, usedNoCache uint64) float64 {
 	}
 	return 0.0
 }
+
+// calculateCPULimit calculate the number of cpus assigned to a container.
+//
+// Calculation is based on 3 alternatives by the following order:
+// - nanocpus:   if set by i.e docker run -cpus=2
+// - cpusetCpus: if set by i.e docker run -docker run -cpuset-cpus="0,2"
+// - cpuquota:   if set by i.e docker run -cpu-quota=50000
+//
+// See https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler for background.
+func calculateCPULimit(hostConfig *ctypes.HostConfig) (float64, error) {
+	var cpuLimit float64
+	var err error
+
+	switch {
+	case hostConfig.NanoCPUs > 0:
+		cpuLimit = float64(hostConfig.NanoCPUs) / 1e9
+	case hostConfig.CpusetCpus != "":
+		cpuLimit, err = parseCPUSet(hostConfig.CpusetCpus)
+		if err != nil {
+			return cpuLimit, err
+		}
+	case hostConfig.CPUQuota > 0:
+		period := hostConfig.CPUPeriod
+		if period == 0 {
+			period = 100000 // Default CFS Period
+		}
+		cpuLimit = float64(hostConfig.CPUQuota) / float64(period)
+	}
+	return cpuLimit, nil
```

Comment:
I would argue that we shouldn't return `nil` here as the error, but return an actual error which says that the CPU limit could not be obtained. As a user, I'd like to know if a metric that I asked for (via configuration) is missing. Otherwise it will be unusual. Maybe at least a warning might be worth it.

Reply:
@gbbr 
Would you think it would be good to return the errors? I don't see this pattern in any other metric and I'm afraid this would create a lot of noise in the logs since it is a common practice not to limit cpu for containers.
What if we added in the metric description something like `This metric is only reported if the container has limits set with -cpus, -cpuset-cpus or -cpu-quota`  
WDYT?

Reply:
I agree. I think that's a good idea.